File: 1701-average-waiting-time/1701-average-waiting-time.cpp
class Solution {
public:
    double averageWaitingTime(vector<vector<int>>& customers) {
        double waitingSum = 0;
        int curTime = 0;
        
        for(auto& customer : customers) {
            waitingSum += max(0, curTime - customer[0]) + customer[1];
            curTime = max(curTime, customer[0]) + customer[1];
        }
        
        return waitingSum / (double) customers.size();
    }
};

File: 1701-average-waiting-time/1701-average-waiting-time.py
class Solution:
    def averageWaitingTime(self, customers: List[List[int]]) -> float:
        waitingList = list()
        curTime = 0
        
        for arrival, time in customers:
            waitingList.append(max(0, curTime - arrival) + time)
            # print("waiting:", max(0, curTime - arrival), "+", time)
            curTime = max(curTime, arrival) + time
        
        return sum(waitingList) / len(waitingList)
            
            
            